src/thx/Iterables.hx

```diff
@@ -106,14 +106,14 @@ of its signature.
 /**
 Refer to `Array.map`.
 **/
-  inline public static function map<T, S>(it : Iterable<T>, f : T -> S) : Array<S>
-    return Iterators.map(it.iterator(), f);
+  inline public static function map<T, S>(it : Iterable<T>, f : T -> S) : Iterable<S>
```

Comment:
if we want to make this change we should also consider if there are other methods (there should be) that transform iterable/iterator into arrays and change them as well?